rakelib/tests.rake

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+# rubocop:disable Lint/SuppressedException
+begin
+  require 'rspec/core/rake_task'
+
+  namespace :tests do
+    desc "Run all RSpec tests"
+    RSpec::Core::RakeTask.new(:spec)
+
+    desc "Run RSpec tests that do not require VM fixtures or a particular shell"
+    RSpec::Core::RakeTask.new(:unit) do |t|
+      t.rspec_opts = '--tag ~ssh --tag ~docker --tag ~bash --tag ~winrm ' \
+                     '--tag ~windows_agents --tag ~puppetserver --tag ~puppetdb ' \
+                     '--tag ~omi --tag ~kerberos'
+    end
+
+    desc 'Run tests that require a host System Under Test configured with WinRM'
+    RSpec::Core::RakeTask.new(:winrm) do |t|
+      t.rspec_opts = '--tag winrm'
+    end
+
+    desc 'Run tests that require a host System Under Test configured with SSH'
+    RSpec::Core::RakeTask.new(:ssh) do |t|
+      t.rspec_opts = '--tag ssh'
+    end
+
+    desc 'Run tests that require a host System Under Test configured with Docker'
+    RSpec::Core::RakeTask.new(:docker) do |t|
+      t.rspec_opts = '--tag docker'
+    end
+
+    desc 'Run tests that require Bash on the local host'
+    RSpec::Core::RakeTask.new(:bash) do |t|
+      t.rspec_opts = '--tag bash'
+    end
+
+    desc 'Run tests that require Windows on the local host'
+    RSpec::Core::RakeTask.new(:windows) do |t|
+      t.rspec_opts = '--tag windows'
+    end
+
+    desc 'Run tests that require OMI docker container'
+    RSpec::Core::RakeTask.new(:omi) do |t|
+      t.rspec_opts = '--tag omi'
+    end
+  end
+
+  # The following tasks are run during CI and require additional environment setup
+  # to run. Jobs that run these tests can be viewed in .github/workflows/
+  namespace :ci do
+    namespace :linux do
+      # Run RSpec tests that do not require WinRM
+      desc ''
```

Comment:
Left the description empty so these don't show up in the task list when running `rake -T`, since these are used for CI instead of local testing. Don't feel strongly one way or the other if we wanted them to show up in the task list, though.